js/util/evented.js

```diff
@@ -78,7 +78,14 @@ var Evented = {
      * @returns {Object} `this`
      */
     fire: function(type, data) {
-        if (!this.listens(type)) return this;
+        if (!this.listens(type)) {
+            // Error events are important for debugging and should not be
+            // silently dropped.
+            if (util.endsWith(type, 'error')) {
```

Comment:
For example, some error events might include a style-spec [`ValidationError` object](https://github.com/mapbox/mapbox-gl-style-spec/blob/mb -pages/lib/error/validation_error.js). Ideally this would be a subclass of `Error` but subclassing `Error` in Javascript requires some hacks.